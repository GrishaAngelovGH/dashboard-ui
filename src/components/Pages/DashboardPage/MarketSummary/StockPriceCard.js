import Card from 'antd/lib/card'
import './StockPriceCard.scss'

const StockPriceCard = ({ title, value, currency, pricing }) => (
    <Card
        className='stock-price-card'
        title={<span className='text-muted'>NASDAQ</span>}
    >
        <h2>{title}</h2>
        <div><span className='h2 font-weight-bold'>{value}</span> {currency}</div>
        <h5 className='text-success font-weight-bold'>{pricing}</h5>
    </Card>
)

export default StockPriceCard