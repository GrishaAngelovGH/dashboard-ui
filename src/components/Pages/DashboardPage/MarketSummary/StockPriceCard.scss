.stock-price-card-container {
    height: 300px;
    color: "#fff";
    line-height: 300px;
    text-align: center;
    background: #007bff;
    display: flex;
    justify-content: center;
    align-items: center;
}

.stock-price-card {
    width: 300px;
    height: 200px;
    border-radius: 5px;
}
