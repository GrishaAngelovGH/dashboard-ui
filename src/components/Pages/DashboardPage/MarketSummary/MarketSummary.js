import { Carousel } from 'antd'
import StockPriceCard from './StockPriceCard'
import './StockPriceCard.scss'

const MarketSummary = ({ stockPrices }) => (
    <div className='row no-gutters'>
        <div className="col-md-12">
            <h2 className='text-center'>Market Summary</h2>

            <Carousel autoplay>
                {
                    stockPrices.map(v => (
                        <div key={v.id}>
                            <div className='stock-price-card-container'>
                                <StockPriceCard
                                    title={v.title}
                                    value={v.value}
                                    currency={v.currency}
                                    pricing={v.pricing}
                                />
                            </div>
                        </div>
                    ))
                }
            </Carousel>
        </div>
    </div>
)

export default MarketSummary