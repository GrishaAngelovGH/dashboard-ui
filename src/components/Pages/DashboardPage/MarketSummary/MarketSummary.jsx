import { Carousel } from 'antd'
import StockPriceCard from './StockPriceCard'
import './StockPriceCard.scss'

const MarketSummary = ({ stockPrices }) => (
	<div className='row g-0'>
		<div className='col-md-12'>
			<h2 className='text-center'>Market Summary</h2>

			<div style={{ width: '800px' }}>
				<Carousel autoplay>
					{
						stockPrices.map(v => (
							<div key={v.id}>
								<div className='stock-price-card-container'>
									<StockPriceCard
										title={v.title}
										value={v.value}
										currency={v.currency}
										pricing={v.pricing}
									/>
								</div>
							</div>
						))
					}
				</Carousel>
			</div>
		</div>
	</div>
)

export default MarketSummary