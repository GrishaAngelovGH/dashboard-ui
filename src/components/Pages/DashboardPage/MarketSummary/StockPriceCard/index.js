import StockPriceCard from './StockPriceCard'

export default StockPriceCard