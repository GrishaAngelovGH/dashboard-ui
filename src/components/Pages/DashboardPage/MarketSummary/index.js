import MarketSummary from './MarketSummary'

export default MarketSummary