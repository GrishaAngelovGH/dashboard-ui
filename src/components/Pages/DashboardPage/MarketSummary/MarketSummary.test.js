import { render } from '@testing-library/react'

import MarketSummary from './MarketSummary'

test('should render component', () => {
    const stockPrices = [
        { id: 1, title: 'Title 1', value: '173,07', currency: 'USD', pricing: '+0.88 (0.51%)' },
        { id: 2, title: 'Title 2', value: '2,789.61', currency: 'USD', pricing: '+17.87 (0.64%)' },
    ]

    const container = render(<MarketSummary stockPrices={stockPrices} />)

    expect(container).toMatchSnapshot()
})