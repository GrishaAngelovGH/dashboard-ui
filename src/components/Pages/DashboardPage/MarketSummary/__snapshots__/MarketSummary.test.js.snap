// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MarketSummary component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row no-gutters"
      >
        <div
          class="col-md-12"
        >
          <h2
            class="text-center"
          >
            Market Summary
          </h2>
          <div
            class="ant-carousel"
          >
            <div
              class="slick-slider slick-initialized"
              dir="ltr"
            >
              <div
                class="slick-list"
              >
                <div
                  class="slick-track"
                  style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                >
                  <div
                    aria-hidden="true"
                    class="slick-slide slick-cloned"
                    data-index="-1"
                    style="width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <div
                          class="stock-price-card-container"
                        >
                          <div
                            class="ant-card ant-card-bordered stock-price-card"
                          >
                            <div
                              class="ant-card-head"
                            >
                              <div
                                class="ant-card-head-wrapper"
                              >
                                <div
                                  class="ant-card-head-title"
                                >
                                  <span
                                    class="text-muted"
                                  >
                                    NASDAQ
                                  </span>
                                </div>
                              </div>
                            </div>
                            <div
                              class="ant-card-body"
                            >
                              <h2>
                                Title 2
                              </h2>
                              <div>
                                <span
                                  class="h2 font-weight-bold"
                                >
                                  2,789.61
                                </span>
                                 
                                USD
                              </div>
                              <h5
                                class="text-success font-weight-bold"
                              >
                                +17.87 (0.64%)
                              </h5>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="false"
                    class="slick-slide slick-active slick-current"
                    data-index="0"
                    style="outline: none; width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <div
                          class="stock-price-card-container"
                        >
                          <div
                            class="ant-card ant-card-bordered stock-price-card"
                          >
                            <div
                              class="ant-card-head"
                            >
                              <div
                                class="ant-card-head-wrapper"
                              >
                                <div
                                  class="ant-card-head-title"
                                >
                                  <span
                                    class="text-muted"
                                  >
                                    NASDAQ
                                  </span>
                                </div>
                              </div>
                            </div>
                            <div
                              class="ant-card-body"
                            >
                              <h2>
                                Title 1
                              </h2>
                              <div>
                                <span
                                  class="h2 font-weight-bold"
                                >
                                  173,07
                                </span>
                                 
                                USD
                              </div>
                              <h5
                                class="text-success font-weight-bold"
                              >
                                +0.88 (0.51%)
                              </h5>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide"
                    data-index="1"
                    style="outline: none; width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <div
                          class="stock-price-card-container"
                        >
                          <div
                            class="ant-card ant-card-bordered stock-price-card"
                          >
                            <div
                              class="ant-card-head"
                            >
                              <div
                                class="ant-card-head-wrapper"
                              >
                                <div
                                  class="ant-card-head-title"
                                >
                                  <span
                                    class="text-muted"
                                  >
                                    NASDAQ
                                  </span>
                                </div>
                              </div>
                            </div>
                            <div
                              class="ant-card-body"
                            >
                              <h2>
                                Title 2
                              </h2>
                              <div>
                                <span
                                  class="h2 font-weight-bold"
                                >
                                  2,789.61
                                </span>
                                 
                                USD
                              </div>
                              <h5
                                class="text-success font-weight-bold"
                              >
                                +17.87 (0.64%)
                              </h5>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide slick-cloned"
                    data-index="2"
                    style="width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <div
                          class="stock-price-card-container"
                        >
                          <div
                            class="ant-card ant-card-bordered stock-price-card"
                          >
                            <div
                              class="ant-card-head"
                            >
                              <div
                                class="ant-card-head-wrapper"
                              >
                                <div
                                  class="ant-card-head-title"
                                >
                                  <span
                                    class="text-muted"
                                  >
                                    NASDAQ
                                  </span>
                                </div>
                              </div>
                            </div>
                            <div
                              class="ant-card-body"
                            >
                              <h2>
                                Title 1
                              </h2>
                              <div>
                                <span
                                  class="h2 font-weight-bold"
                                >
                                  173,07
                                </span>
                                 
                                USD
                              </div>
                              <h5
                                class="text-success font-weight-bold"
                              >
                                +0.88 (0.51%)
                              </h5>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide slick-cloned"
                    data-index="3"
                    style="width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <div
                          class="stock-price-card-container"
                        >
                          <div
                            class="ant-card ant-card-bordered stock-price-card"
                          >
                            <div
                              class="ant-card-head"
                            >
                              <div
                                class="ant-card-head-wrapper"
                              >
                                <div
                                  class="ant-card-head-title"
                                >
                                  <span
                                    class="text-muted"
                                  >
                                    NASDAQ
                                  </span>
                                </div>
                              </div>
                            </div>
                            <div
                              class="ant-card-body"
                            >
                              <h2>
                                Title 2
                              </h2>
                              <div>
                                <span
                                  class="h2 font-weight-bold"
                                >
                                  2,789.61
                                </span>
                                 
                                USD
                              </div>
                              <h5
                                class="text-success font-weight-bold"
                              >
                                +17.87 (0.64%)
                              </h5>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <ul
                class="slick-dots slick-dots-bottom"
                style="display: block;"
              >
                <li
                  class="slick-active"
                >
                  <button>
                    1
                  </button>
                </li>
                <li
                  class=""
                >
                  <button>
                    2
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row no-gutters"
    >
      <div
        class="col-md-12"
      >
        <h2
          class="text-center"
        >
          Market Summary
        </h2>
        <div
          class="ant-carousel"
        >
          <div
            class="slick-slider slick-initialized"
            dir="ltr"
          >
            <div
              class="slick-list"
            >
              <div
                class="slick-track"
                style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="-1"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="stock-price-card-container"
                      >
                        <div
                          class="ant-card ant-card-bordered stock-price-card"
                        >
                          <div
                            class="ant-card-head"
                          >
                            <div
                              class="ant-card-head-wrapper"
                            >
                              <div
                                class="ant-card-head-title"
                              >
                                <span
                                  class="text-muted"
                                >
                                  NASDAQ
                                </span>
                              </div>
                            </div>
                          </div>
                          <div
                            class="ant-card-body"
                          >
                            <h2>
                              Title 2
                            </h2>
                            <div>
                              <span
                                class="h2 font-weight-bold"
                              >
                                2,789.61
                              </span>
                               
                              USD
                            </div>
                            <h5
                              class="text-success font-weight-bold"
                            >
                              +17.87 (0.64%)
                            </h5>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active slick-current"
                  data-index="0"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="stock-price-card-container"
                      >
                        <div
                          class="ant-card ant-card-bordered stock-price-card"
                        >
                          <div
                            class="ant-card-head"
                          >
                            <div
                              class="ant-card-head-wrapper"
                            >
                              <div
                                class="ant-card-head-title"
                              >
                                <span
                                  class="text-muted"
                                >
                                  NASDAQ
                                </span>
                              </div>
                            </div>
                          </div>
                          <div
                            class="ant-card-body"
                          >
                            <h2>
                              Title 1
                            </h2>
                            <div>
                              <span
                                class="h2 font-weight-bold"
                              >
                                173,07
                              </span>
                               
                              USD
                            </div>
                            <h5
                              class="text-success font-weight-bold"
                            >
                              +0.88 (0.51%)
                            </h5>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide"
                  data-index="1"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="stock-price-card-container"
                      >
                        <div
                          class="ant-card ant-card-bordered stock-price-card"
                        >
                          <div
                            class="ant-card-head"
                          >
                            <div
                              class="ant-card-head-wrapper"
                            >
                              <div
                                class="ant-card-head-title"
                              >
                                <span
                                  class="text-muted"
                                >
                                  NASDAQ
                                </span>
                              </div>
                            </div>
                          </div>
                          <div
                            class="ant-card-body"
                          >
                            <h2>
                              Title 2
                            </h2>
                            <div>
                              <span
                                class="h2 font-weight-bold"
                              >
                                2,789.61
                              </span>
                               
                              USD
                            </div>
                            <h5
                              class="text-success font-weight-bold"
                            >
                              +17.87 (0.64%)
                            </h5>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="2"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="stock-price-card-container"
                      >
                        <div
                          class="ant-card ant-card-bordered stock-price-card"
                        >
                          <div
                            class="ant-card-head"
                          >
                            <div
                              class="ant-card-head-wrapper"
                            >
                              <div
                                class="ant-card-head-title"
                              >
                                <span
                                  class="text-muted"
                                >
                                  NASDAQ
                                </span>
                              </div>
                            </div>
                          </div>
                          <div
                            class="ant-card-body"
                          >
                            <h2>
                              Title 1
                            </h2>
                            <div>
                              <span
                                class="h2 font-weight-bold"
                              >
                                173,07
                              </span>
                               
                              USD
                            </div>
                            <h5
                              class="text-success font-weight-bold"
                            >
                              +0.88 (0.51%)
                            </h5>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="3"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="stock-price-card-container"
                      >
                        <div
                          class="ant-card ant-card-bordered stock-price-card"
                        >
                          <div
                            class="ant-card-head"
                          >
                            <div
                              class="ant-card-head-wrapper"
                            >
                              <div
                                class="ant-card-head-title"
                              >
                                <span
                                  class="text-muted"
                                >
                                  NASDAQ
                                </span>
                              </div>
                            </div>
                          </div>
                          <div
                            class="ant-card-body"
                          >
                            <h2>
                              Title 2
                            </h2>
                            <div>
                              <span
                                class="h2 font-weight-bold"
                              >
                                2,789.61
                              </span>
                               
                              USD
                            </div>
                            <h5
                              class="text-success font-weight-bold"
                            >
                              +17.87 (0.64%)
                            </h5>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <ul
              class="slick-dots slick-dots-bottom"
              style="display: block;"
            >
              <li
                class="slick-active"
              >
                <button>
                  1
                </button>
              </li>
              <li
                class=""
              >
                <button>
                  2
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
